### PR TITLE
Feat: adding more comment at config.yaml

### LINKED ISSUE #254: Proposal: Adding more comments at example/config.yaml
## Description

See the https://github.com/merico-dev/stream/issues/252#issuecomment-1056938530 for more info.

### PR DESCRIPTION
**Pull request by daniel-hutao**
Signed-off-by: Daniel Hu <tao.hu@merico.dev>

# Summary

Feat: adding more comment at config.yaml

### Related Issues

close #254

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -147,7 +147,11 @@ make e2e
 
 ## Configuration
 
-See [examples/config.yaml](./examples/config.yaml).
+There is a default configuration in [examples/config.yaml](./examples/config.yaml).
```
Comment (IronCore864) [inline README.md:150]: "default" means even if you don't change anything, it should work.

I think "This is an example of DevStream config" is more suitable here.

File: README.md [hunk 2]
```diff
@@ -147,7 +147,11 @@ make e2e
...
+
+Remember to open this configuration file, modify all **DO_CHANGE_ME** in it to your own.
+
+Pay attention to the meaning of each item to ensure that it is you want.
```
Comment (IronCore864) [inline README.md:154]: "that it is you want" -> "that it is what you want"

I highly recommend the usage of Grammarly for documentation update.

File: examples/config.yaml [hunk 3]
```diff
@@ -1,116 +1,215 @@
...
       chart_name: jenkins/jenkins
+      # release name of the chart
       release_name: dev
+      # k8s namespace where jenkins will be installed
       namespace: jenkins
+      # whether to wait for the release to be deployed or not
       wait: true
+      # the time to wait for any individual Kubernetes operation (like Jobs for hooks). This defaults to 5m0s
       timeout: 5m
+      # whether to perform a CRD upgrade during installation
```
Comment (IronCore864) [inline examples/config.yaml:214]: As the OP suggested, this `examples/config.yaml` file has grown too big for new users to get a "quick start". 

As a workaround, maybe we can create multiple "examples", like:

- `examples/golang-webapp-ci-cd.yaml`: github repo scaffolding + github actions + argocd + argocd app
- `examples/golang-webapp-trello.yaml`: github repo scaffolding + github actions + trello
- `examples/observability.yaml`: devlake + prometheus
- `examples/gitlab-ci.yaml`: gitlab ci (TODO: here maybe we can add gitlab repo scaffolding later as well)
- `examples/jenkins.yaml`: jenkins

Credit to @daniel-hutao who suggested that we should have some "scenarios" for users to begin quickly.

**Review by IronCore864 — CHANGES_REQUESTED** [review]

**Comment by daniel-hutao** [comment]
@IronCore864 Please take a review again.

**Review by IronCore864 — APPROVED** [review]

**Review by lfbdev — APPROVED** [review]
/lgtm

**Pull request merged**